pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -475,6 +530,86 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *api.Ser
 		status.Ingress = append(status.Ingress, api.LoadBalancerIngress{IP: floatIP.FloatingIP})
 	}
 
+	if lbaas.opts.ManageSecurityGroups {
+
+		lbSecGroupCreateOpts := groups.CreateOpts{
+			Name:        getSecurityGroupName(clusterName, apiService),
+			Description: fmt.Sprintf("Securty Group for %v Service LoadBalancer", apiService.Name),
+		}
+
+		lbSecGroup, err := groups.Create(lbaas.network, lbSecGroupCreateOpts).Extract()
+
+		if err != nil {
+			return nil, err
```

Comment:
This, and all other early returns below should unwind and cleanup the loadbalancer.
```go
_ = lbaas.EnsureLoadBalancerDeleted(clusterName, apiService)
return nil, err
```